### DISCUSSION TITLE
Has anyone gotten the path claimed error after DiFFoZ's fix?

### DISCUSSION THREAD

**Discussion by SDGNelson** [SmartlyDressedGames/Unturned-3.x-Community#4061] [General]
The most recent update included a proposed fix by DiFFoZ explained here: https://github.com/SmartlyDressedGames/Unturned-3.x-Community/issues/2884#issuecomment-1651142290

I'm optimistic that may have been the problem all along. If so, it was 100% my fault, young Nelson hacked that code in without a second thought back in 2014 🤦‍♂️, and updating A*PFP wouldn't have fixed it.

That being said, I would still like to replace pathfinding at some point with a map-wide navmesh rather than the per-location navmeshes.

**Comment by DiFFoZ** [comment]
My patch did help a bit, but now new errors occurring.

One of server network reported me this error:
```
[2023-08-12 02:00:35] Exception: Trying to initialize a node when it is not safe to initialize any nodes. Must be done during a graph update
[2023-08-12 02:00:35] AstarPath.InitializeNode (Pathfinding.GraphNode node) (at <79f203b2e95e4ad99df7f800b9c30dd6>:0)
Pathfinding.GraphNode..ctor (AstarPath astar) (at <79f203b2e95e4ad99df7f800b9c30dd6>:0)
Pathfinding.MeshNode..ctor (AstarPath astar) (at <79f203b2e95e4ad99df7f800b9c30dd6>:0)
Pathfinding.TriangleMeshNode..ctor (AstarPath astar) (at <79f203b2e95e4ad99df7f800b9c30dd6>:0)
SDG.Unturned.LevelNavigation.buildGraph (SDG.Unturned.River river) (at <da7afc137a6d4041a2cef668152379d7>:0)
SDG.Unturned.LevelNavigation.load () (at <da7afc137a6d4041a2cef668152379d7>:0)
SDG.Unturned.Level+<init>d__136.MoveNext () (at <da7afc137a6d4041a2cef668152379d7>:0)
UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) (at <966ecbc21bec4138a5192b5e1b2dd41f>:0)
UnityEngine.MonoBehaviour:StartCoroutine(IEnumerator)
SDG.Unturned.Level:onSceneLoaded(Scene, LoadSceneMode)
UnityEngine.SceneManagement.SceneManager:Internal_SceneLoaded(Scene, LoadSceneMode)
```

I already fixed it in my patcher v1.0.2 by patching `LevelNavigation.load()` under `if (Provider.isServer)` check:
https://github.com/DiFFoZ/AStarPatcher/blob/master/AStarPatcher/Patches/Patch_LevelNavigation.cs#L70

And also in v1.0.1 I fixed a bug in Astar lib. `LockFreeStack` was using `Interlocked` on `Path` and it's incorrect. I replaced it with `lock` instead https://github.com/DiFFoZ/AStarPatcher/blob/master/AStarPatcher/Patches/Patch_LockFreeStack.cs
I also checked archived versions of Astar lib, and it was fixed in `3.8` release (Unturned uses `3.5.9.1` beta(?) version)

**Reply by SDGNelson** [reply]
Interesting, thanks for your insights! It sounds like young-Nelson strikes again. Would you be comfortable / allow me implementing the same change in the vanilla code?

**Reply by DiFFoZ** [reply]
yes, sure!
